### ISSUE TITLE
allow mixing --project command line parameter with a list of files, includes, and exclludes

### ISSUE THREAD

**Issue by zpdDG4gta8XKpMCd** [microsoft/TypeScript#12958]
`--project` command line argument should be allowed with an explicit list of files as well as `includes` list and `excludes` lists

command line arguments should take precedence over parameters in config, just like params of `compilerOptions`

`{ files }` vs --files // --files wins
`{ includes }` vs --includes // --includes wins
...
etc

**Comment by aleksey-bykov** [comment]
### case 1: multiple entry points to the application

I have 11 entry points (topmost scripts to start execution from), so i need to generate 11 different bundles using `outFile` which requires me to create and support 11 different configs
```
tsconfig.a.json
tsconfig.b.json
tsconfig.c.json
tsconfig.d.json
tsconfig.e.json
tsconfig.f.json
tsconfig.g.json
tsconfig.h.json
tsconfig.i.json
tsconfig.j.json
tsconfig.k.json
tsconfig.l.json
tsconfig.m.json
tsconfig.o.json
```

and we are planning to add 27 more

**Comment by aleksey-bykov** [comment]
case 2: compiling a sub-tree of all files based on `compilerOptions` from `tsconfig.json`

another use case, for the sake of saving some time i to compile a only sub-tree of our 1500+ project starting at given root file, yet using the `compilerOptions` from the `tsconfig.json`

**Comment by timocov** [comment]
Related https://github.com/Microsoft/TypeScript/issues/29241#issuecomment-459789340

TL;DR: `paths` compiler option cannot be passed via CLI.

**Comment by RyanCavanaugh** [comment]
Let's track at #27379
